sql/src/main/java/org/apache/druid/sql/calcite/expression/builtin/ArrayOverlapOperatorConversion.java

```diff
@@ -59,11 +59,13 @@ public class ArrayOverlapOperatorConversion extends BaseExpressionDimFilterOpera
               "'ARRAY_OVERLAP(array, array)'",
               OperandTypes.or(
                   OperandTypes.family(SqlTypeFamily.ARRAY),
-                  OperandTypes.family(SqlTypeFamily.STRING)
+                  OperandTypes.family(SqlTypeFamily.STRING),
+                  OperandTypes.family(SqlTypeFamily.NUMERIC)
```

Comment:
it seems like its possible to possible to specify operands with different types
could you add some tests like:
```
SELECT dim3 FROM druid.numfoo WHERE ARRAY_OVERLAP(l1||'.1', ARRAY[2.1, 7.1]) LIMIT 5
SELECT dim3 FROM druid.numfoo WHERE ARRAY_OVERLAP(l1||'.1', ARRAY[2.1, 7.1]) LIMIT 5
SELECT dim3 FROM druid.numfoo WHERE ARRAY_OVERLAP(ARRAY[l2], ARRAY[2, 7]) LIMIT 5
SELECT dim3 FROM druid.numfoo WHERE ARRAY_OVERLAP(ARRAY['x',l2||''], ARRAY['2', '7']) LIMIT 5
SELECT dim3 FROM druid.numfoo WHERE ARRAY_OVERLAP(ARRAY[2,7], ARRAY['2', '7']) LIMIT 5
SELECT dim3 FROM druid.numfoo WHERE ARRAY_OVERLAP(ARRAY[2,7], ARRAY[2.0,7.0]) LIMIT 5
```

it seems like type mismatch was already there in some cases (`array<int>,array<string>)` - would it be possible to reject invalid cases?

Reply:
These are being taken care of in [this PR](https://github.com/apache/druid/pull/15974)
But rather than rejecting them, we would cast them properly and return correct response

Reply:
I don't explicitly see these cases being covered in that PR (or I missed it).
Can we cover them somehow?